adapters/timm/timm_model_adapter.py

```diff
@@ -0,0 +1,358 @@
+
+
+@dl.Package.decorators.module(name='model-adapter',
+                              description='Model Adapter for TIMM Models',
+                              init_inputs={'model_entity': dl.Model})
+class TIMMAdapter(dl.BaseModelAdapter):
+    """
+    MobileNetV3 Model adapter using Pytorch.
+    The class bind Dataloop model and model entities with model code implementation
+    """
+    def __init__(self, model_entity=None):
+        self.device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+        self.model_name = self.configuration.get('model_name')
+        super(TIMMAdapter, self).__init__(model_entity=model_entity)
+
+    def load(self, local_path, **kwargs):
+        """ Loads model and populates self.model with a `runnable` model
+
+            This function is called by load_from_model (download to local and then loads)
+
+        :param local_path: `str` directory path in local FileSystem
+        """
+        weights_filename = self.model_entity.configuration.get('weights_filename')
+        model_path = str(os.path.join(local_path, weights_filename))
+        if weights_filename and os.path.exists(model_path):
+            # With the model entity and weights from the platform
+
+            logger.info(f"Loading a model from {model_path}")
+            self.model = torch.load(model_path, map_location=self.device)
+            self.model.to(self.device)
+            self.model.eval()
+            logger.info(f"Loaded model from {model_path} successfully")
+        else:
+            # With the model from the library
+            self.model = timm.create_model(self.model_name, pretrained=True)
+            self.model.to(self.device)
+            self.model.eval()
+            logger.info(f"Loaded model from library successfully")
+
+    def save(self, local_path, **kwargs):
+        """ Saves configuration and weights locally
+
+            The function is called in save_to_model which first save locally and then uploads to model entity
+
+        :param local_path: `str` directory path in local FileSystem
+        """
+        weights_filename = kwargs.get('weights_filename', 'model.pth')
+        torch.save(self.model, os.path.join(local_path, weights_filename))
+        self.configuration['weights_filename'] = weights_filename
+
+    def predict(self, batch, **kwargs):
+        """ Model inference (predictions) on batch of images
+
+        :param batch: `np.ndarray`
+        :return: `list[dl.AnnotationCollection]` each collection is per each image / item in the batch
+        """
+        input_size = self.configuration.get('input_size', 256)
+
+        preprocess = transforms.Compose([
+            transforms.ToPILImage(),
+            transforms.Resize((input_size, input_size)),
+            transforms.CenterCrop(224),
+            transforms.ToTensor(),
+            transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]),
+        ])
+
```

Comment:
all models is normalized by the same mean and std? i think resent uses different values, maybe it can be an input from the configuration